go/tasks/plugins/k8s/kfoperators/common/common_operator.go

```diff
@@ -0,0 +1,142 @@
+package common
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/tasklog"
+
+	commonOp "github.com/kubeflow/tf-operator/pkg/apis/common/v1"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	flyteerr "github.com/lyft/flyteplugins/go/tasks/errors"
+	"github.com/lyft/flyteplugins/go/tasks/logs"
+	pluginsCore "github.com/lyft/flyteplugins/go/tasks/pluginmachinery/core"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	TensorflowTaskType = "tensorflow"
+	PytorchTaskType    = "pytorch"
+)
+
+func ExtractCurrentCondition(jobConditions []commonOp.JobCondition) (commonOp.JobCondition, error) {
+	if jobConditions != nil {
+		sort.Slice(jobConditions, func(i, j int) bool {
+			return jobConditions[i].LastTransitionTime.Time.After(jobConditions[j].LastTransitionTime.Time)
+		})
+
+		for _, jc := range jobConditions {
+			if jc.Status == v1.ConditionTrue {
+				return jc, nil
+			}
+		}
+	}
+
+	return commonOp.JobCondition{}, fmt.Errorf("found no current condition. Conditions: %+v", jobConditions)
+}
+
+func GetPhaseInfo(currentCondition commonOp.JobCondition, occurredAt time.Time,
+	taskPhaseInfo pluginsCore.TaskInfo) (pluginsCore.PhaseInfo, error) {
+	switch currentCondition.Type {
+	case commonOp.JobCreated:
+		return pluginsCore.PhaseInfoQueued(occurredAt, pluginsCore.DefaultPhaseVersion, "JobCreated"), nil
+	case commonOp.JobRunning:
+		return pluginsCore.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, &taskPhaseInfo), nil
+	case commonOp.JobSucceeded:
+		return pluginsCore.PhaseInfoSuccess(&taskPhaseInfo), nil
+	case commonOp.JobFailed:
+		details := fmt.Sprintf("Job failed:\n\t%v - %v", currentCondition.Reason, currentCondition.Message)
+		return pluginsCore.PhaseInfoRetryableFailure(flyteerr.DownstreamSystemError, details, &taskPhaseInfo), nil
+	case commonOp.JobRestarting:
+		details := fmt.Sprintf("Job failed:\n\t%v - %v", currentCondition.Reason, currentCondition.Message)
+		return pluginsCore.PhaseInfoRetryableFailure(flyteerr.RuntimeFailure, details, &taskPhaseInfo), nil
+	}
+
+	return pluginsCore.PhaseInfoUndefined, nil
+}
+
+func GetLogs(taskType string, name string, namespace string,
+	workersCount int32, psReplicasCount int32, chiefReplicasCount int32) ([]*core.TaskLog, error) {
+	taskLogs := make([]*core.TaskLog, 0, 10)
+
+	logConfig := logs.GetLogConfig()
+	logPlugin, err := logs.InitializeLogPlugins(logs.GetLogConfig())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if logPlugin == nil {
+		return nil, nil
+	}
+
+	if logConfig.IsKubernetesEnabled {
```

Comment:
I missed this in my previous review... I don't think you need this line any more... just call GetTaskLogs (like you do below) and pass as much information as you can... let the log plugin take care of which log links to produce...

Reply:
Sorry, I forgot to remove it. And I just updated it.